### PR TITLE
Added a service to zero the robot's ftsensor

### PR DESCRIPTION
**Pull request by fmauch**
As script code interaction needs some deeper knowledge about the pitfalls of URScript, this MR provides an easy-access service to zero the robot's ftsensor.

Currently, we always advertise the service, I'd like to only advertise it when an e-Series is connected or report `false` when being called on a CB3 robot.

### CODE REVIEW

File: ur_robot_driver/src/ros/hardware_interface.cpp [hunk 1]
```diff
@@ -315,6 +315,19 @@ bool HardwareInterface::init(ros::NodeHandle& root_nh, ros::NodeHandle& robot_hw
   // Calling this service will make the "External Control" program node on the UR-Program return.
   deactivate_srv_ = robot_hw_nh.advertiseService("hand_back_control", &HardwareInterface::stopControl, this);
 
+  // Calling this service will zero the robot's ftsensor. Note: On e-Series robots this will only
+  // work when the robot is in remote-control mode.
+  tare_sensor_srv_ = robot_hw_nh.advertiseService<std_srvs::Trigger::Request, std_srvs::Trigger::Response>(
+      "zero_ftsensor", [&](std_srvs::Trigger::Request& req, std_srvs::Trigger::Response& resp) {
+        resp.success = this->ur_driver_->sendScript(R"(sec tareSensor():
+  set_digital_out(1,True)
```
Comment (gavanderhoorn) [inline ur_robot_driver/src/ros/hardware_interface.cpp]: I'm not entirely up-to-speed on e-Series controllers: why the `set_digital_out(..)` here? Is the ft sensor connected to it?

Comment (fmauch) [reply ur_robot_driver/src/ros/hardware_interface.cpp]: oops, that was from debugging...

Comment (fmauch) [reply ur_robot_driver/src/ros/hardware_interface.cpp]: fixed.

File: ur_robot_driver/src/ros/hardware_interface.cpp [hunk 2]
```diff
@@ -694,6 +698,28 @@ bool HardwareInterface::resendRobotProgram(std_srvs::TriggerRequest& req, std_sr
...
+{
+  if (ur_driver_->getVersion().major < 5)
+  {
+    std::stringstream ss;
+    ss << "Zeroing the Force-Torque sensor is only available for e-Series robots (Major version >= 5). This robot's "
+          "version is "
+       << ur_driver_->getVersion();
+    ROS_ERROR_STREAM(ss.str());
+    res.message = ss.str();
+    res.success = false;
```
Comment (gavanderhoorn) [inline ur_robot_driver/src/ros/hardware_interface.cpp]: As `res.message` is just a `string`, would the following work?

```c++
res.message = "Zeroing the Force-Torque sensor is only available for e-Series robots (Major version >= 5). This robot's version is " + ur_driver_->getVersion();
ROS_ERROR_STREAM(res.message);
```

Avoids creating a `std::stringstream`.

Comment (fmauch) [reply ur_robot_driver/src/ros/hardware_interface.cpp]: Not quite, as `UrDriver::getVersion()` returns an object of type `VersionInformation` which has a `<<` operator (also implemented in this PR). With your suggestion we would have to add a `toString()` method or similar to `VersionInformation` instead. Personally, I like using stream operators for this. Is there any particular reason why you would prefer avoiding the `std::stringstream`?

Comment (gavanderhoorn) [reply ur_robot_driver/src/ros/hardware_interface.cpp]: It's a pattern I've used before. Hence the suggestion.

I'm not against using `std::stringstream` here.

**Pull request merged**